pkg/db/query.go

```diff
@@ -0,0 +1,250 @@
+	args := []interface{}{datestamp}
+
+	// Check if we need to filter by component and update query if so
+	if components != nil {
+		// Convert components to an interface slice
+		iComponents := make([]interface{}, len(components))
+		for i := range components {
+			iComponents[i] = components[i]
+		}
+		// Append the query and arguments together
+		query, args = appendQueryConditional(query, args, "AND component = Any(ARRAY[%v])", iComponents)
+	}
+
+	//log.Println("DEBUG - EXECUTING QUERY:")
+	//log.Println(query)
+	//log.Println(args)
+	rows, err := c.database.Query(query, args...)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	// Parse the data into bugzilla.Bugs
+	var b bugzilla.Bug
+	var bugs []bugzilla.Bug
+
+	/*// DEBUGGING
+	ct, _ := rows.ColumnTypes()
+	cc, _ := rows.Columns()
+	for ind, column := range cc {
+		log.Printf("column %s[%v] is of type %#v", column, ind, ct[ind].ScanType().Name())
+	}*/
+	for rows.Next() {
+		err = rows.Scan(
+			&b.Id,
+			&b.Component,
+			&b.TargetRelease,
+			&b.AssignedTo,
+			&b.Status,
+			&b.Summary,
+			&b.Keywords,
+			&b.PmScore,
+			&b.Externals,
+			&b.DateStamp,
+		)
+		if err != nil {
+			log.Printf("Error scanning row: %v", err)
+		} else {
+			bugs = append(bugs, b)
+		}
+	}
+	err = rows.Err()
+	if err != nil {
+		return nil, fmt.Errorf("error while scanning rows of bugs: %v", err)
+	}
+
+	return bugs, nil
+}
+
+// Breakdown represents the totals for all, new, and closed bugs in the query
+// TODO - where should Breakdown actually live...? or should we return 3 ints?
+type Breakdown struct {
+	Total  int
+	New    int
+	Closed int
+}
+
+// GetBreakdown calculates the counts for total bugs, new bugs, and closed bugs
+func (c postgresClient) GetBreakdown(startDate, endDate string, components []string, keywords []string, custCase bool, targetRelease string) (Breakdown, error) {
+	var total int
+	var new int
+	var closed int
+
+	// Setup the base query
+	query := "SELECT COUNT(id) FROM bugs WHERE datestamp = $1"
+	args := []interface{}{endDate}
+
+	// QUERY CRAFTING
+	// TODO - switch/case instead of ifs?
+	if components != nil {
+		// Convert components to an interface slice
+		iComponents := make([]interface{}, len(components))
+		for i := range components {
+			iComponents[i] = components[i]
+		}
+		// Append the query and arguments together
+		query, args = appendQueryConditional(query, args, "AND component = Any(ARRAY[%v])", iComponents)
+	}
+	if len(keywords) > 0 {
+		// Convert keywords to an interface slice
+		iKeywords := make([]interface{}, len(keywords))
+		for i := range keywords {
+			iKeywords[i] = keywords[i]
+		}
+		// Append the query and arguments together
+		query, args = appendQueryConditional(query, args, "AND keywords && ARRAY[%v]", iKeywords)
+	}
+	// TODO - if custCase {appendQuery "json stuff"}
+	if custCase == true {
+		// Query the jsonb directly
+		// NOTE - If json structure changes, this may break.  Look at simple string search.
```

Comment:
if json structure changes, you're going to have other issues.  This is fine.  String search brings its own issues.